users/serializers.py

```diff
@@ -21,28 +23,24 @@ class RegistrationSerializer(serializers.ModelSerializer):
         error_messages = {
             "required": "Email field is required.", 
             "invalid" : "Kindly enter a Valid Email Address",
-        })
-
-    def save(self):
```

Comment:
### I found a new :bug: :beetle: . It goes like this:
1. I created a new user:
![image](https://user-images.githubusercontent.com/33046846/82069646-3fce7080-96f1-11ea-9833-236349641a32.png)
1. Pasted the same credentials to login api
![image](https://user-images.githubusercontent.com/33046846/82069716-5aa0e500-96f1-11ea-82c9-18eeeb1827c6.png)
1. Yet I get `wrong credentials 401`
![image](https://user-images.githubusercontent.com/33046846/82069766-6db3b500-96f1-11ea-94a4-3a797187e390.png)

### Please find the solution. [hint]: This bug is not present in the dev branch.

Reply:
I expect these:
1. Find the bug
1. fix it
1. How you will prevent it from happening again?

Reply:
I see that I accidentally removed the save function while checking on errors. This was preventing the data to be saved in the database, and since the user wasn't present in the database and the entry was valid, it gave 401.

To fix this I've added the save function. 

To prevent this I've added a test to the valid submission, so as to check whether registered user is being accepted at login or not. The test is working fine in my local.
```
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
.....................
----------------------------------------------------------------------
Ran 21 tests in 0.946s

OK
Destroying test database for alias 'default'...
```

Reply:
@priyanshisharma nice work :clap: . You nearly found the actually bug. The actual bug is:
1. when save() function is not overloaded it uses the User.objects.create() function. 
1. In our overloaded save() we use User.objects.create_user().
1. The difference is create_user encodes the password which the authenticate()  used in login api is assuming. When create() is used the unencoded password doesnot match with the actual user.
1. one additional detail here is: in we dont override save() then we wont set the email as username. which will infact create `username not unique 500 error` :bug: 

an encoded password will look like this:
`pbkdf2_sha256$180000$7iKPU6y7680n$v456wvx2KhDvgDgboAVg57rMpOFOZv6bXk2r8BR/Kjs=`

Reply:
you can resolve this conversation when u see it

Reply:
Got it! 💯💯 
Thank you!😄